lib/events/event_handler.ex

```diff
@@ -56,6 +59,27 @@ defmodule ElixirTools.Events.EventHandler do
     :ok
   end
 
+  @spec publish(Event.t(), event_schema, [events_opt]) :: :ok
+  def publish(event, schema, opts) do
+    event_module = opts[:event_module] || Event
+    task_supervisor_module = opts[:task_supervisor_module] || Task.Supervisor
+    not_sent_event_module = opts[:not_sent_event_module] || NotSentEvent
+
+    case event_module.validate_json_schema(schema, event) do
+      :ok ->
+        task_supervisor_module.async_nolink(ElixirTools.TaskSupervisor, fn ->
+          publish_event_call(event, opts)
+        end)
+
+      {:error, reason} ->
+        error_info = event |> Map.from_struct() |> Map.put(:reason, reason)
+
+        not_sent_event_module.create!(%{content: Jason.encode!(error_info)})
```

Comment:
`create!` function is from Schema directly - so not really needed to inject `not_sent_event_module`

Reply:
I'll remove it and replace the test check to check Ecto table.